common/interfaces/user_interface/ui_events.c

```diff
@@ -0,0 +1,149 @@
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "ui_events.h"
+
+#include <string.h>
+
+#include "memzero.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+static ui_event_status_t event_status;
+static ui_event_t event;
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+void ui_status_mark_ready_for_events() {
+  memzero(&event, sizeof(event));
+  event_status = UI_EVENT_SCR_RENDERED;
+}
+
+void ui_get_events(ui_event_t *ui_event_os_obj) {
+  memcpy(ui_event_os_obj, &event, sizeof(ui_event_t));
+}
+
+void ui_status_reset_event_state() {
+  memzero(&event, sizeof(event));
+  event_status = UI_EVENT_NONE;
+}
+
+bool ui_set_confirm_event() {
+  if (event_status != UI_EVENT_SCR_RENDERED)
+    return false;    // Invalid state return
+
+  event.event_occured = SEC_TRUE;
+  event.event_type = UI_EVENT_CONFIRM;
+}
+
+bool ui_set_cancel_event() {
+  if (event_status != UI_EVENT_SCR_RENDERED)
+    return false;    // Invalid state return
+
+  event.event_occured = SEC_TRUE;
+  event.event_type = UI_EVENT_REJECT;
+}
+
+bool ui_set_list_event(uint8_t list_selection) {
+  if (event_status != UI_EVENT_SCR_RENDERED)
+    return false;    // Invalid state return
+
+  event.event_occured = SEC_TRUE;
+  event.event_type = UI_EVENT_LIST_CHOICE;
+  event.list_selection = list_selection;
+}
```

Comment:
These were added as gaurds of rare situations to prevent a consecutive events to overwrite the earlier one.

As discussed, we have concluded that this will not be the case as ui screen destructors will be called just after setting an event.
We can remove the statuses as they do not add any gaurds with the current UI implementation.